### ISSUE TITLE
<pb> interaction in MIDI file generation/first <pb> missed in SVG generation

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#713]
MIDI file output from verovio is messed up by `<pb>` elements.  The current time in the score is reset to 0 when a `pb` element is encountered.

Test data:

<img width="889" alt="screen shot 2017-10-22 at 8 21 42 am" src="https://user-images.githubusercontent.com/3487289/31863395-0d4b2eae-b702-11e7-802d-2f6fb6c8fb2c.png">

A page break is added after each measure in the MEI data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-10-22T08:11:36" version="2.0.0-dev-ae70647-dirty">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000000668601322" midi.bpm="400">
                        <staffGrp xml:id="staffgrp-0000000708315607">
                            <staffDef xml:id="staffdef-0000001158551528" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000000097660845">
                        <measure xml:id="measure-L3" n="1">
                            <staff xml:id="staff-L3F1N1" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <note xml:id="note-L4F1" dur="1" oct="4" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <pb n="2"/>
                        <measure xml:id="measure-L5" n="2">
                            <staff xml:id="staff-L5F1N1" n="1">
                                <layer xml:id="layer-L5F1N1" n="1">
                                    <note xml:id="note-L6F1" dur="2" oct="4" pname="d" accid.ges="n" />
                                    <note xml:id="note-L7F1" dur="2" oct="4" pname="e" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <pb n="3"/>
                        <measure xml:id="measure-L8" n="3">
                            <staff xml:id="staff-L8F1N1" n="1">
                                <layer xml:id="layer-L8F1N1" n="1">
                                    <note xml:id="note-L9F1" dur="4" oct="4" pname="f" accid.ges="n" />
                                    <note xml:id="note-L10F1" dur="4" oct="4" pname="g" accid.ges="n" />
                                    <note xml:id="note-L11F1" dur="4" oct="4" pname="a" accid.ges="n" />
                                    <note xml:id="note-L12F1" dur="4" oct="4" pname="b" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <pb n="4"/>
                        <measure xml:id="measure-L13" right="end" n="4">
                            <staff xml:id="staff-L13F1N1" n="1">
                                <layer xml:id="layer-L13F1N1" n="1">
                                    <note xml:id="note-L14F1" dur="1" oct="4" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

When converting this data to a MIDI file on the command line:

```bash
verovio -t midi file.mei
```

The resulting MIDI file plays all measures at the same time, resulting in a MIDI performance which is 1/4 as long as it should be.

Converting with `--no-layout` option:

```bash
verovio -t midi file.mei --no-layout
```

The MIDI file generation is correct.

-------------

There is also a second bug which is related:   verovio ignores the first page break in a file (at least when there are no system breaks).  This will influence the MIDI output in the enscripten version of the toolkit, but not in the command-line version.  In other words, the first two pages of MIDI output will be correct in enscription, but this not true on the command-line (where the MIDI time will be reset to zero after every `pb`).   This may indicate an uninitialized variable in the C++ class that deals with `pb`.

Here is an example of the above test file in VHV (type alt-m to switch to MEI mode, and then copy-and-paste the test MEI contents into the editor).

The first page contains two pages instead of the expected one:

![screen shot 2017-10-22 at 8 35 06 am](https://user-images.githubusercontent.com/3487289/31863589-422f88f2-b704-11e7-8228-7c2445da8da7.png)

The rest of the pages are as expected:

![screen shot 2017-10-22 at 8 35 08 am](https://user-images.githubusercontent.com/3487289/31863591-4f09e126-b704-11e7-88c6-d31da3f14ab8.png)

![screen shot 2017-10-22 at 8 35 11 am](https://user-images.githubusercontent.com/3487289/31863593-52378fd8-b704-11e7-8755-af213ab0b8b7.png)

When pressing the "play" button in VHV, the first two measures will play correctly, then starting with the second page, both the 3rd and 4th measures play at the same time.

This bug is also one reason for the problem reported in issue https://github.com/rism-ch/verovio/issues/712:  The first two pages of music are squeezed onto one page. `<pb n="2"/>` is placed after measure 6 in the Hummel score, but the first page in issue #712 has 11 measures.  Likewise when playing MIDI with the Hummel score, the first two page play correctly, then the rest of the score is collapsed into one measure.  When converting on the command-line to MIDI, all measures (including those on the first page), are collapsed into a single measure.

**Comment by craigsapp** [comment]
This is related to issue #638.   Either there has been a regression, and/or there is some uninitialized variable which is causing transient behavior.

**Comment by lpugin** [comment]
Fixed dfff3ad0448eb7b2540c8272ebd265255c1d4467
